logrotate/Options.cmake

```diff
@@ -9,27 +9,50 @@
 #
 # example: cmake -DBUILD_STATIC=ON .. 
 option (BUILD_STATIC "Building g3logrotate as static library. Linking with static g3log library" OFF)
-IF (BUILD_STATIC) 
+IF (BUILD_STATIC)
   SET(G3LOG_LIBRARY_TYPE g3logger)
-  SET(G3LOG_LIBRARY_NAME libg3logger.a)
   MESSAGE("'OPTION: cmake -DBUILD_STATIC=ON'\nLinking statically with g3log. Creating static library for g3logrotate")
+  
+  IF (MSVC)
+    SET(G3LOG_LIBRARY_NAME_DEBUG g3logger_d.lib)
+	SET(G3LOG_LIBRARY_NAME_RELEASE g3logger.lib)
```

Comment:
Wrote this quite in a rush. Should have read it again prior to posting it.

I hope though that it kinda makes sense
.
So g3sink debug will require glog debug.
Similarly g3sink release will require g3log release.

This means that both have to be defined, unless you have a better way to
sort this out.
This is the exact same way that you have sort it out boost which also
requires you to specify both release and debug version.

The cmake for this is kinda cludgy, so once you specify dynamic linking
then its hard to go back to static unless you delete the cmake cache.
So long as you stick to one thing though it will work. Cmake will go
through the errors and step by step all necessary info is set (at least
using cmake gui).
If you set everything with -D in cmd lilne then it should be OK but have
not tested it.

I did recently though add g3log to our teamcity server along with
publishing nuget package to our internal feed and that worked pretty nicely
using cmake configure and build.

On 19 December 2015 at 00:45, Christos Stamatopoulos <
xstamatopoulos@gmail.com> wrote:

> for example if you want to generate g3sink